### PR TITLE
Add a scheduler for reindexing

### PR DESCRIPTION
**Pull request by marko-bekhta**
fixes https://github.com/quarkusio/search.quarkus.io/issues/80

### CODE REVIEW

File: src/test/java/io/quarkus/search/app/indexing/SchedulerTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@QuarkusIOSample.Setup(filter = QuarkusIOSample.SearchServiceFilterDefinition.class)
+class SchedulerTest {
+    public static class Profile implements QuarkusTestProfile {
+        @Override
+        public Map<String, String> getConfigOverrides() {
+            return Map.of("indexing.on-startup.enabled", "false",
+                    "indexing.scheduled.enabled", "true",
+                    // every 1 minute:
+                    "indexing.scheduled.cron", "0 0/1 * ? * * *");
```
Comment (yrodiere) [inline src/test/java/io/quarkus/search/app/indexing/SchedulerTest.java:39]: Would something lower (e.g. 5s, or 15s) make sense?
I'd rather limit the impact on the duration of the test suite...

File: src/main/java/io/quarkus/search/app/indexing/IndexingConfig.java [hunk 2]
```diff
@@ -9,11 +9,22 @@
...
         @WithDefault("3s")
         Duration waitInterval();
     }
+
+    interface Scheduled {
+        @WithDefault("true")
+        boolean enabled();
+
+        // At 19:00:00 every day
+        @WithDefault("0 0 19 * * ?")
```
Comment (yrodiere) [inline src/main/java/io/quarkus/search/app/indexing/IndexingConfig.java:27]: This probably makes more sense to put in `application.properties`, given how specific it is?
You can have properties without defaults.

File: src/main/java/io/quarkus/search/app/indexing/IndexingService.java [hunk 3]
```diff
@@ -86,6 +87,23 @@ void indexOnStartup(@Observes StartupEvent ev) {
                         t -> Log.errorf(t, "Reindexing on startup failed: %s", t.getMessage()));
     }
 
+    @Scheduled(cron = "{indexing.scheduled.cron}")
+    void indexOnTime() {
+        if (!indexingConfig.scheduled().enabled()) {
+            Log.debug("Scheduled reindex is disabled...");
+            return;
+        }
```
Comment (yrodiere) [inline src/main/java/io/quarkus/search/app/indexing/IndexingService.java:95]: This seems unnecessary, as setting `indexing.scheduled.cron=off` would also disable scheduled indexing. See javadoc of `io.quarkus.scheduler.Scheduled#cron`.

**Review by yrodiere — APPROVED** [review]
Thanks! LGTM, just a few suggestions, but this can be merged as is if you prefer.

**Comment by marko-bekhta** [comment]
thanks, I've tried it with the smaller interval and it worked out ok (at least in the local build). I've used +5sec delay just so that we don't see unnecessary exceptions logged in the build 🙈😃

**Review by yrodiere — APPROVED** [review]

**Pull request merged**